CrossBrowserTestingTunnel.js

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * A CrossBrowserTesting tunnel.
+ *
+ * @constructor module:digdug/CrossBrowserTestingTunnel
+ * @extends module:digdug/Tunnel
+ */
+function CrossBrowserTestingTunnel() {
+	this.apikey = process.env.CBT_APIKEY;
+	this.username = process.env.CBT_USERNAME;
+	this.servers = [];
+	Tunnel.apply(this, arguments);
+}
+
+var _super = Tunnel.prototype;
+CrossBrowserTestingTunnel.prototype = util.mixin(Object.create(_super), /** @lends module:digdug/CrossBrowserTestingTunnel# */ {
+	constructor: CrossBrowserTestingTunnel,
+
+	/**
+	 * The CrossBrowserTesting API key.
+	 *
+	 * @type {string}
+	 * @default the value of the CBT_APIKEY environment variable
+	 */
+	apikey: null,
+
+	/**
+	 * The CrossBrowserTesting username.
+	 *
+	 * @type {string}
+	 * @default the value of the CBT_USERNAME environment variable
+	 */
+	username: null,
+
+	directory: pathUtil.join(__dirname, 'CrossBrowserTesting'),
+
+	executable: 'java',
+
+	/**
+	 * A list of regular expressions corresponding to domains whose connections should fail immediately if the VM
+	 * attempts to make a connection to them.
+	 *
+	 * @type {string[]}
+	 */
+	servers: null,
+
+	/**
+	 * A filename where additional logs from the tunnel should be output.
+	 *
+	 * @type {string}
+	 */
+	logFile: null,
+
+	hostname: 'hub.crossbrowsertesting.com',
+
+	port: 80,
+
+	url: 'https://github.com/crossbrowsertesting/cbt-tunnel-java/raw/master/cbttunnel.jar',
+
+	/**
+	 * Whether or not to use rabbIT compression for the tunnel connection.
+	 *
+	 * @type {boolean}
+	 * @default
+	 */
+	useCompression: false,
+
+	/**
+	 * Whether or not to use the default local Jetty proxy for the tunnel.
+	 *
+	 * @type {boolean}
+	 * @default
+	 */
+	useJettyProxy: true,
+
+	/**
+	 * Whether or not to use the default remote Squid proxy for the VM.
+	 *
+	 * @type {boolean}
+	 * @default
+	 */
+	useSquidProxy: true,
+
+	/**
+	 * Whether or not to re-encrypt data encrypted by self-signed certificates.
+	 *
+	 * @type {boolean}
+	 * @default
+	 */
+	useSsl: false,
+
+	get auth() {
+		return this.apikey + ':' + this.username;
+	},
+
+	get isDownloaded() {
+		return fs.existsSync(pathUtil.join(this.directory, 'CrossBrowserTesting-tunnel/cbttunnel.jar'));
+	},
+
+	_makeArgs: function (readyFile) {
+		var args = [
+			'-jar', 'CrossBrowserTesting-tunnel/cbttunnel.jar',
+			this.apikey,
+			this.username,
+			'-P', this.port,
+			'-f', readyFile
+		];
+
+		this.servers.length && args.push('-F', this.servers.join(','));
+		this.logFile && args.push('-l', this.logFile);
+		this.useJettyProxy || args.push('-x');
+		this.useSquidProxy || args.push('-q');
+		this.useCompression && args.push('-b');
+		this.useSsl && args.push('-s');
+		this.verbose && args.push('-d');
+
+		return args;
+	},
+
+	sendJobState: function (jobId, data) {
+		var payload = JSON.stringify({
+			status: data.status || data.success ? 'completed' : 'error'
+		});
+
+		return request.put('https://api.CrossBrowserTesting.com/v1/tests/' + jobId, {
```

Comment:
From digging through their API, the current address for job state seems to be `https://crossbrowsertesting.com/api/v3/selenium/`